core/domain/state_domain.py

```diff
@@ -1777,15 +1777,6 @@ def _validate_drag_and_drop_input(self, strict: bool = False) -> None:
                             ele_position = ele['position']
                             ele_element = ele['element']
 
-                            if ele_position > len(rule_spec.inputs['x']):
```

Comment:
Why are we removing this check?

Reply:
We would require to add the `continue` part here, mainly the validation error from here was removed because it was not a valid case. Consider the below case:
```
Choices: 1, 2, and 3

Equals [[1, 2, 3], [4]]
HasElement 1 at position 3
All Other Answers: ...
```
Now here the position 3 does not exists but it is possible that user have placed the element at position 3 and we should not raise an error at that time.